dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/ws/WSBatchAuthService.java

```diff
@@ -10,168 +10,175 @@
+
+    /**
+     * Returns the given instance id if the current user is authorized to stop it.
+     *
+     * @param instanceId the instanceId to stop
+     * @return the given instanceId that is authorized to be stopped
+     * @throws JobSecurityException if the current user does not have authority to stop the given instanceId
+     */
+    public abstract long authorizedJobStopByInstance(long instanceId) throws JobSecurityException;
+
+    /**
+     * Returns the given instance id if the current user is authorized to purge it
+     *
+     * @param instanceId the job instanceId to purge
+     * @return
+     * @throws NoSuchJobInstanceException
+     * @throws JobSecurityException f the current user does not have authority to purge the given instanceId
+     */
+    public long authorizedJobPurgeByInstance(long instanceId) throws NoSuchJobInstanceException, JobSecurityException;
+
+    /**
+     * @return true if the current user is part of the batchAdmin role, false otherwise
+     */
+    public abstract boolean isAdmin();
+
+    /**
+     * @return true if the current user is part of the batchSubmitter role, false otherwise
+     */
+    public abstract boolean isSubmitter();
+
+    /**
+     * @return true if the current user is part of the batchMonitor role, false otherwise
+     */
+    public abstract boolean isMonitor();
+
+    /**
+     * @return true if the current user is part of the batchGroupAdmin role, false otherwise
+     */
+    public boolean isGroupAdmin();
+
+    /**
+     * /**
+     *
+     * @return true if the current user is part of the batchGroupMonitor role, false otherwise
+     */
+    public abstract boolean isGroupMonitor();
+
+    /**
+     * @return true if the current user is part of any bathc role. batchAdmin, batchSubmitter, or batchMonitor, false otherwise
+     */
+    public boolean isInAnyBatchRole();
+
+    /**
+     * Returns the given instance id if the current user is authorized to abandon it
+     *
+     * @param instanceId the job instanceId to abandon
+     * @return
+     * @throws NoSuchJobInstanceException
+     * @throws JobSecurityException f the current user does not have authority to abandon the given instanceId
+     */
+    long authorizedJobAbandonByInstance(long instanceId) throws NoSuchJobInstanceException, JobSecurityException;
+
+    /**
+     * @param stepExecutionId
+     *
+     * @throws IllegalArgumentException if the corresponding step execution entry doesn't exist in the persistent store.
+     * @throws JobSecurityException f the current user does not have authority to read the given stepExecutionId
+     */
+    public long authorizedStepExecutionRead(long stepExecutionId) throws IllegalArgumentException, JobSecurityException;
+
+    /**
+     * @param jobList
+     * @return
+     */
+    public List<WSJobInstance> filterFoundJobInstancesBasedOnGroupSecurity();
+
```

Comment:
Is this **filterFoundJobInstancesBasedOnGroupSecurity** method needed?

Reply:
removed and also from impl